prime-router/src/test/kotlin/azure/BlobAccessTests.kt

```diff
@@ -52,4 +73,237 @@ class BlobAccessTests {
             assertThat(e).isNotNull()
         }
     }
+
+    @Test
+    fun `upload report`() {
+        val testUrl = "http://uploadreport"
+        val testFormat = Report.Format.CSV
+        val testBytes = "testbytes".toByteArray()
+
+        val one = Schema(name = "one", topic = Topic.TEST)
+        val metadata = Metadata(schema = one)
+        val report1 = Report(
+            one, listOf(listOf("1", "2"), listOf("3", "4")), source = TestSource,
+            bodyFormat = testFormat, metadata = metadata
+        )
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every {
+            BlobAccess.uploadBody(
+                report1.bodyFormat, testBytes, report1.name, null,
+                Event.EventAction.NONE
+            )
+        } returns
+            BlobAccess.BlobInfo(report1.bodyFormat, testUrl, BlobAccess.sha256Digest(testBytes))
+
+        val testBlob = BlobAccess()
+        val result = testBlob.uploadReport(report1, testBytes)
+
+        assertThat(result.format).isEqualTo(testFormat)
+        assertThat(result.blobUrl).isEqualTo(testUrl)
+        assertThat(result.digest).isEqualTo(BlobAccess.sha256Digest(testBytes))
+    }
+
+    @Test
+    fun `upload body`() {
+        val testUrl = "http://uploadbody"
+        val testFormat = Report.Format.CSV
+        val testName = "testblob"
+        val testBytes = "testbytes".toByteArray()
+        val testFolder = "testfolder"
+        val testEnv = "testenvvar"
+        val testEvents: List<Event.EventAction?> = listOf(
+            Event.EventAction.RECEIVE,
+            Event.EventAction.SEND,
+            Event.EventAction.BATCH,
+            Event.EventAction.PROCESS,
+            Event.EventAction.ROUTE,
+            Event.EventAction.TRANSLATE,
+            Event.EventAction.NONE,
+            Event.EventAction.CONVERT,
+            null
+        )
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every { BlobAccess.Companion.getBlobConnection(testEnv) } returns "testconnection"
+        every { BlobAccess.Companion.uploadBlob(any(), testBytes) } returns testUrl
+
+        testEvents.forEach {
+            val result = when (it) {
+                null -> BlobAccess.uploadBody(testFormat, testBytes, testName, "")
+                Event.EventAction.CONVERT -> BlobAccess.uploadBody(testFormat, testBytes, testName, action = it)
+                else -> BlobAccess.uploadBody(testFormat, testBytes, testName, testFolder, it)
```

Comment:
Is this arbitrarily to just cover both null and non-null `testFolder` values? If so, a one-line comment to mention that would be helpful for folks reading this test without the context of code coverage in mind.

Reply:
I added a comment to clarify this.